File: nom-api/Nom.Orch/Models/Household/HouseholdInviteTokenResponseModel.cs
// File: Nom.Orch/Models/Household/HouseholdInviteTokenResponseModel.cs

namespace Nom.Orch.Models.Household
{
    public class HouseholdInviteTokenResponseModel
    {
        public long Id { get; set; }
        public long HouseholdId { get; set; }
        public string Token { get; set; } = string.Empty;
        public DateTime CreatedDate { get; set; }
    }
} 

File: nom-ui/src/app/household/models/household-invite-token-response.model.ts
// File: nom-ui/src/app/household/models/household-invite-token-response.model.ts

import { IHouseholdInviteTokenResponseModel } from './household-invite-token-response.model.interface';

export class HouseholdInviteTokenResponseModel implements IHouseholdInviteTokenResponseModel {
    id = 0;
    householdId = 0;
    token = '';
    expiresAt?: Date;
    createdDate: Date = new Date();
    usesLeft?: number;

    constructor(data?: Partial<IHouseholdInviteTokenResponseModel>) {
        if (data) {
            Object.assign(this, data);
        }
    }
} 